docs/source/tune_cli.rst

```diff
@@ -39,19 +39,25 @@ to download files using the CLI.
 Download a model
 ----------------
 
-The ``tune download <path>`` command downloads any model from the Hugging Face Hub.
+The ``tune download <path>`` command downloads any model from the Hugging Face or Kaggle Model Hub.
 
 .. list-table::
    :widths: 30 60
 
    * - \--output-dir
-     - Directory in which to save the model.
+     - Directory in which to save the model. Note: this is option not yet supported when `--source` is set to `kaggle`.
    * - \--output-dir-use-symlinks
      - To be used with `output-dir`. If set to 'auto', the cache directory will be used and the file will be either duplicated or symlinked to the local directory depending on its size. It set to `True`, a symlink will be created, no matter the file size. If set to `False`, the file will either be duplicated from cache (if already exists) or downloaded from the Hub and not cached.
    * - \--hf-token
      - Hugging Face API token. Needed for gated models like Llama.
    * - \--ignore-patterns
      - If provided, files matching any of the patterns are not downloaded. Defaults to ignoring safetensors files to avoid downloading duplicate weights.
+   * - \--source {huggingface,kaggle}
+     - If provided, downloads model weights from the provided <path> on the designated source hub.
+   * - \--kaggle-username
```

Comment:
I wasn't sure if this would add more confusion for the developer experience, considering `kagglehub.login()` is only invoked from Python but `tune download` is a CLI. Granted, notebooks incorporate both. I'm curious for your thoughts on that though.

Reply:
After more thought, I'm not sure `kagglehub.login()` is viable here. It will only set the credentials within the running Python context, and will not modify the environment variables to authenticate across contexts. So even if a user calls in somewhere in their jupyter/kaggle notebook, those credentials will not carry over to the shell command `tune download`.

cc @rosbo for your take here

Reply:
That's correct, `kagglehub.login()` doesn't persist the credentials on "disk".

Reply:
Got it - so the user would have to explicitly set their credentials as environment variables with e.g. `export KAGGLE_TOKEN=asdkfasdfasdfasdf`, is that correct?

Reply:
The reason I'm asking is that the username is easy to remember and pass in again and again, but the token is only visible once and can be sensitive to just store in a note. I might prefer that we add a note recommending they save it to their `.bash_profile` or something.

Reply:
> so the user would have to explicitly set their credentials as environment variables with e.g. `export KAGGLE_TOKEN=asdkfasdfasdfasdf`

yes, that's correct (with a small note - the necessary env vars are `KAGGLE_KEY` **and** `KAGGLE_USERNAME`).

> but the token is only visible once and can be sensitive to just store in a note

agreed. I think it's safer to push users towards using environment variables. In this case, I was mostly going for consistency with the `--hf-token` argument.

Reply:
just to make sure I understand the immediate suggestion. should we:

1. completely _remove_ the `--kaggle-username` and `--kaggle-api-key` arguments, and/or 2. opt for an error message to set the env vars in the users shell profile

Reply:
also, another alternative to consider: we could add a `--kaggle-login` argument instead. when present, `torchtune` would call `kagglehub.login()` before running `kagglehub.model_download()`.

Reply:
> just to make sure I understand the immediate suggestion. should we:
> 
> 1. completely _remove_ the `--kaggle-username` and `--kaggle-api-key` arguments, and/or
> 2. opt for an error message to set the env vars in the users shell profile

Keep the username and api key arguments. And if they are NOT empty, log a "tip" that the user can store these in their environment instead of passing these arguments in every time.

Reply:
Resolved here: c2dee3a6911a48fa718fb40face90aa190cabb9b